app/src/main/java/com/tari/android/wallet/ui/activity/CreateWalletActivity.kt

```diff
@@ -0,0 +1,208 @@
+    var mainContentShiftMargin = 0
+
+    override val contentViewId = R.layout.activity_create_wallet
+
+    private var videoLoopCount = 0
+    private val videoLoopHandler = Handler()
+    private val uiHandler = Handler()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out)
+        init()
+    }
+
+    override fun onStart() {
+        super.onStart()
+        videoView1.setVideoURI(getResourceUri(R.raw.splash_video_loop_1))
+        videoView2.setVideoURI(getResourceUri(R.raw.splash_video_loop_2))
+        videoView1.setOnPreparedListener {
+            // setVideoLoopNotifier(videoView1.duration)
+        }
+        videoView1.start()
+    }
+
+    override fun onStop() {
+        super.onStop()
+        videoLoopHandler.removeCallbacksAndMessages(null)
+        uiHandler.removeCallbacksAndMessages(null)
+    }
+
+    private val videoLoopRunnable: Runnable = Runnable {
+        videoLoopCount = if (++videoLoopCount < 2) videoLoopCount else 0
+
+        if (videoLoopCount == 0) {
+            videoView2.pause()
+            videoView1.seekTo(2000)
+            videoView1.start()
+        } else {
+            videoView1.pause()
+            videoView2.seekTo(2000)
+            videoView2.start()
+        }
+
+        setVideoLoopNotifier(videoView1.duration)
+    }
+
+    private fun setVideoLoopNotifier(duration: Int) {
+        if (duration == -1) return
+        videoLoopHandler.removeCallbacks(videoLoopRunnable)
+        videoLoopHandler.postDelayed(videoLoopRunnable, (duration - 3000).toLong())
+    }
+
+    private fun init() {
+        rootView.post { startViewAnimation() }
+        setVideoDimension()
+    }
+
+    private fun setVideoDimension() {
+        rootView.viewTreeObserver.addOnGlobalLayoutListener(
+            object : ViewTreeObserver.OnGlobalLayoutListener {
+                override fun onGlobalLayout() {
+                    rootView.viewTreeObserver.removeOnGlobalLayoutListener(this)
+                    val rootWidth = rootView.width
+                    val viewHeight =
+                        cryptoWalletMainText1.top - cryptoWalletMainText1.height - mainContentShiftMargin
+                    val viewWidth = rootWidth.coerceAtMost(viewHeight) - tariAnimationView.top
+                    val lp: FrameLayout.LayoutParams =
+                        videoContainer.layoutParams as FrameLayout.LayoutParams
+
+                    lp.width = viewWidth
+                    lp.height = viewHeight - videoContainer.marginTop
+                    videoContainer.layoutParams = lp
+                }
+            })
+    }
+
+    @OnClick(R.id.create_wallet_btn_layout)
+    fun onCreateWalletClick() {
+        createWalletButton.visibility = View.GONE
+        progressBar.visibility = View.VISIBLE
+        uiHandler.postDelayed({
+            startTariViewAnimation()
+        }, 2000)
```

Comment:
Please avoid magic numbers, you can constants instead either in Constants.kt or in the file.